adhocracy4/comments_async/static/comments_async/comment.jsx

```diff
@@ -198,16 +198,18 @@ export default class Comment extends React.Component {
       } else {
         content = this.props.children
       }
-      comment = (
-        <div className={'a4-comments__text' + (this.state.anchored ? ' a4-comments__text--highlighted' : '')}>
-          <ReactMarkdown
-            disallowedTypes={['heading']}
-            plugins={[[gfm, { singleTilde: false }]]}
-          >
-            {content}
-          </ReactMarkdown>
-        </div>
-      )
+      if (!this.props.is_blocked) {
```

Comment:
I am a bit confused about all the ifs and elses here. 
But anyway, what is this for? Why do we need that if? Where is the difference to is_removed and is_censored?

Reply:
Quite frankly, I don't know what the point of having is_censored and is_blocked is since I agree that these seem to mean the same thing. But since they are seperate properties of comments, I presume there is or was an intent of having both. is_removed is different since the writer of the comment can remove it, and this the shows a different label than if a moderator removes a comment:
![image](https://user-images.githubusercontent.com/17978375/140743411-52a7df9f-75a2-46b2-8446-c1736d808491.png)

Reply:
Yes, I know that they are nearly the same. (We discussed why we added the is_blocked instead of using is_censored.)
My question was why you need the if-statement here for is_blocked that you do not need for is_censored. So, this if statement is only for is_blocked. Why did you add it?

Reply:
Oh, this is because while is_censored deletes the text of the comment, is_blocked doesn't delete the text, it only stops it from being shown. Does that make sense?

Reply:
Yes! But we also do not show the text for the deleted comments (you changed that in your PR fixing how the deleted comments are shown, right?). So, do we change the text in the serializer? Or how do we manage to not show the text for censored or removed comments?
Right, or did you change that in the method? Yes, that's what we did! Hmm, then maybe instead of this if-statement we do change it in the serializer? Or make the whole if-bit more readable. 
I would prefer changing the serializer. What do you think @Rineee and @phillimorland ?

Reply:
as we already use the serializer to remove the user image and name I think we should change it in the serializer too, it's more consistent with our other code

Reply:
Ok, then:
Do I overwrite a get-method for the comment text itself in the comments_async/serializer to return the comment(-text) only if it is not is_blocked? 
Then I could get rid of this if-statement, but just to make sure I'm getting the method right...

Reply:
Yes, that's how you do it! You add it as a SerializerMethodField and tell it in the `get_comment` what to do.

Reply:
Ok, I've figured out how to change it and pushed this. But 4 tests are failing due to it having a negative impact on the comments (i.e. child comments made after the blocking are also blocked automatically). So I still have to solve this.